src/controller/datacontroller.js

```diff
@@ -198,10 +198,28 @@ export default class DataController {
 			this.model.selection.removeAllRanges();
 			this.model.selection.clearAttributes();
 
+			// Parse data to model and extract markers from parsed document fragment.
+			const { modelItem, markersData } = this.parse( data );
```

Comment:
Creating the real `MarkersCollection` on `DocumentFragment` might be more compilated because `Marker` keeps `LiveRange` inside but `LiveRange` work only for `Document`. We can:
- try to improve `DocumentFragment` and `LiveRange` to work each other but it might be overengineering. 
- don't create the real `MarkersCollection` on `DocumentFragment`but only something to store markers data. I don't like it because every change on DocFrag will make markers positions out of date.